src/goose/brokers/redis/batch.clj

```diff
@@ -70,3 +60,72 @@
        :retrying    retrying
        :successful  successful
        :dead        dead})))
+
+(defn set-batch-expiration
+  [conn id linger-in-hours]
+  (let [batch-state-key (d/prefix-batch id)
+        enqueued-job-set (d/construct-batch-job-set id d/enqueued-job-set)
+        retrying-job-set (d/construct-batch-job-set id d/retrying-job-set)
+        successful-job-set (d/construct-batch-job-set id d/successful-job-set)
+        dead-job-set (d/construct-batch-job-set id d/dead-job-set)
+        linger-in-sec (u/hour->sec linger-in-hours)]
+    (redis-cmds/expire conn batch-state-key linger-in-sec)
+    (redis-cmds/expire conn enqueued-job-set linger-in-sec)
+    (redis-cmds/expire conn retrying-job-set linger-in-sec)
+    (redis-cmds/expire conn successful-job-set linger-in-sec)
+    (redis-cmds/expire conn dead-job-set linger-in-sec)))
+
+(defn post-batch-exec
+  [conn id]
+  (car-locks/with-lock conn "post-batch-exec"
+                       d/redis-batch-lock-timeout-ms
+                       d/redis-batch-lock-wait-ms
+                       (let [{:keys [batch-state] :as batch} (get-batch-state conn id)
+                             status (batch/status-from-counts batch)
+                             {:keys [callback-job-id
+                                     callback-fn-sym
+                                     linger-in-hours
+                                     queue
+                                     ready-queue
+                                     retry-opts]} batch-state]
+                         (when (= status batch/status-complete)
+                           (when (and
+                                   (some? callback-fn-sym)
+                                   (not callback-job-id))
+                             (let [callback-args (-> (select-keys batch [:successful :dead])
+                                                     (assoc :id id)
```

Comment:
suggestion: do we want to give id as part of the map?
Could it be:
`callback [id map] ()`
This map can expand in the future to have other things

Reply:
Makes sense, id can be the first arg. Made the change.